docs/google/best_practices.md

```diff
@@ -4,7 +4,28 @@ This section lists some best practices for creating a circuit that performs well
 on Google hardware devices. This is an area of active research, so users are
 encouraged to try multiple approaches to improve results.
 
-## Use built-in optimizers as a first pass
+This guide is split into three parts:
+*  Getting your circuit to run
+*  Lowering error
+*  Making it run faster
+
+
+## Getting a circuit to run on hardware
+
+In order to run on hardware, the circuit must only use qubits and gates that the
+device supports.  Using inactive qubits, non-adjacent qubits, or non-native
+gates will immediately cause a circuit to fail.
+
+Attaching a `cirq.Device` to a circuit will test a lot of these conditions for
```

Comment:
Is this something we want to lean in to? I always considered the fact that circuits have a device to be a vestigial design choice that no one uses in practice and IMO doesn't make much sense. From my point of view, it would make sense for devices to validate circuits: `cirq.google.Sycamore.validate_circuit(circuit)`. You could even imagine a scheme where instead of just raising an exception at the first problem it could return a structured list of problems and suggestions on how to fix them.

Reply:
Is this something we are trying to get rid of?  I guess either way doesn't matter much to me.  I changed it to validate using the device.

Reply:
opened https://github.com/quantumlib/Cirq/issues/3295 to discuss the broader question